.github/workflows/ci-cd.yml

```diff
@@ -32,10 +35,34 @@ jobs:
           echo "MSYS2 packages: ${msys2_packages}"
           echo "::set-output name=msys2_packages::${msys2_packages}"
 
+      - name: Detect changed files
```

Comment:
You can combine these tasks with a ternary operator as such:
`run: git diff --name-only "${{ github.event.before ? github.event.before : github.base_ref }}..HEAD" > _changed_files`.

I think there may be an even more elegant way, but we can try that later.

Reply:
If the interpreter GitHub uses is based on JavaScript, `github.event.before || github.base_ref` could work.
That said, the current code makes it obvious why we're using two different fields. This version is shorter but doesn't tell us the same thing.
Also, if GitHub starts sending the `before` field in the PR payload with a semantically different value, this logic might break. It's probably very unlikely, though.

Reply:
Looks like this doesn't actually work. A little more trickery is required. Nonetheless, there is no need for two tasks.

Reply:
This should be fine now.